core-plugins/src/e2e-test/features/joiner/Joiner.feature

```diff
@@ -0,0 +1,75 @@
+    When Select plugin: "File" from the plugins list as: "Source"
+    When Expand Plugin group in the LHS plugins list: "Analytics"
+    When Select plugin: "Joiner" from the plugins list as: "Analytics"
+    Then Connect plugins: "File" and "Joiner" to establish connection
+    Then Connect plugins: "File2" and "Joiner" to establish connection
+    Then Move plugins: "File" by xOffset 200 and yOffset 200
+    When Expand Plugin group in the LHS plugins list: "Sink"
+    When Select plugin: "File" from the plugins list as: "Sink"
+    Then Connect plugins: "Joiner" and "File3" to establish connection
+    Then Navigate to the properties page of plugin: "File"
+    Then Enter input plugin property: "referenceName" with value: "firstName"
+    Then Enter input plugin property: "path" with value: "gcsSourceBucket1"
+    Then Select dropdown plugin property: "format" with option value: "csv"
+    Then Click plugin property: "skipHeader"
+    Then Click plugin property: "enableQuotedValues"
+    Then Click on the Get Schema button
+    Then Verify the Output Schema matches the Expected Schema: "firstNameOutputSchema"
+    Then Validate "File" plugin properties
+    Then Close the Plugin Properties page
+    Then Navigate to the properties page of plugin: "File2"
+    Then Enter input plugin property: "referenceName" with value: "lastName"
+    Then Enter input plugin property: "path" with value: "gcsSourceBucket2"
+    Then Select dropdown plugin property: "format" with option value: "csv"
+    Then Click plugin property: "skipHeader"
+    Then Click plugin property: "enableQuotedValues"
+    Then Click on the Get Schema button
+    Then Verify the Output Schema matches the Expected Schema: "lastNameOutputSchema"
+    Then Validate "File" plugin properties
+    Then Close the Plugin Properties page
+    Then Navigate to the properties page of plugin: "Joiner"
+    Then Expand fields
+    Then Uncheck plugin "File2" field "id" alias checkbox
+    Then Enter numPartitions 1
+    Then Validate "Joiner" plugin properties
+    Then Close the Plugin Properties page
+    Then Navigate to the properties page of plugin: "File3"
+    Then Enter input plugin property: "referenceName" with value: "result"
+    Then Enter input plugin property: "path" with value: "gcsTargetBucket"
+    Then Select dropdown plugin property: "format" with option value: "csv"
+    Then Close the Plugin Properties page
+    Then Save the pipeline
+    Then Preview and run the pipeline
+    Then Wait till pipeline preview is in running state
+    Then Open and capture pipeline preview logs
+    Then Verify the preview run status of pipeline in the logs is "succeeded"
+    Then Close the pipeline logs
+    Then Close the preview
+    Then Deploy the pipeline
+    Then Run the Pipeline in Runtime
+    Then Wait till pipeline is in running state
+    Then Open and capture logs
+    Then Verify the pipeline status is "Succeeded"
+    Then Close the pipeline logs
```

Comment:
can we verify that the output written to the file sink is correct?

Reply:
good question. But the data in GCS will be partitioned and hard to verify even if I [set the number of partitions to 1](https://github.com/cdapio/hydrator-plugins/pull/1645/files/e6a093058ecf689af2221190cb2835d13502f8fa#diff-395a661e25a14cf563e72940dad92a122606550bae42774737d4a8bb3f89bae2R55) in `Joiner`

Reply:
ah I see. is there a reason we want to write to GCS sink? If you write to BQ you it will be easier verify

Reply:
Can we compose all the files in GCS under the particular destination bucket and then read and verify?
https://stackoverflow.com/questions/24187243/how-to-use-compose-on-gcs-using-the-java-client